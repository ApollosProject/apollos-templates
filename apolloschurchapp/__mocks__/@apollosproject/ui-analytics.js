import React from 'react';

module.exports = {
  track: () => '',
  AnalyticsConsumer: ({ children }) => children({ test: jest.fn() }),
  AnalyticsProvider: ({ children }) => children,
  CoreNavigationAnalytics: ({ children }) =>
    children({ onNavigationStateChange: jest.fn() }),
  TrackEventWhenLoaded: () => null,
  withTrackOnPress: (Component) => (props) => <Component {...props} />,
};
