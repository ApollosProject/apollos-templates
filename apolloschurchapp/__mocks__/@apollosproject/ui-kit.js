module.exports = {
  ...jest.requireActual('@apollosproject/ui-kit'),
  NavigationService: {
    navigate: jest.fn(),
    setTopLevelNavigator: jest.fn(),
  },
};
