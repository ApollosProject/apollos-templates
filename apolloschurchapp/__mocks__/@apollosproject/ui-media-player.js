module.exports = {
  MediaPlayerSpacer: ({ children }) => children,
  MediaPlayer: () => 'MediaPlayer',
  MediaPlayerProvider: ({ children }) => children,
  playVideoMutation: 'mutation { playVideo }',
  withTabBarMediaSpacer: () => ({ children }) => children,
};
