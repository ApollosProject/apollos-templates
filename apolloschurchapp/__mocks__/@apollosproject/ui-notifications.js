module.exports = { NotificationsProvider: ({ children }) => children };
