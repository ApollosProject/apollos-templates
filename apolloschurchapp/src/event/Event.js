import React from 'react';
import { get } from 'lodash';
import PropTypes from 'prop-types';
import moment from 'moment';
import InAppBrowser from 'react-native-inappbrowser-reborn';
import {
  GradientOverlayImage,
  BackgroundView,
  PaddedView,
  H2,
  Paragraph,
  StretchyView,
} from '@apollosproject/ui-kit';
import HTMLView from '@apollosproject/ui-htmlview';

import { FlexedScrollView, EventInfoItem } from './components';

const handlePressAnchor = (url) => {
  InAppBrowser.open(url);
};

const Event = ({ event, loading }) => {
  const coverImageSources = get(event, 'image.sources', []);
  return (
    <BackgroundView>
      <StretchyView>
        {({ Stretchy, ...scrollViewProps }) => (
          <FlexedScrollView {...scrollViewProps}>
            {coverImageSources.length || loading ? (
              <Stretchy>
                <GradientOverlayImage
                  isLoading={!coverImageSources.length && loading}
                  source={coverImageSources}
                />
              </Stretchy>
            ) : null}
            <PaddedView vertical={false}>
              <H2 padded isLoading={!event.name && loading}>
                {event.name}
              </H2>
              <Paragraph>
                <EventInfoItem
                  icon={'time'}
                  title={moment(event.start).format('ddd, MMMM Do, YYYY')}
                  subtitle={`${moment(event.start).format('LT')} â€” ${moment(
                    event.end
                  ).format('LT')}`}
                />
                <EventInfoItem icon={'pin'} title={event.location} />
              </Paragraph>
              <HTMLView isLoading={loading} onPressAnchor={handlePressAnchor}>
                {event.description}
              </HTMLView>
            </PaddedView>
          </FlexedScrollView>
        )}
      </StretchyView>
    </BackgroundView>
  );
};

Event.propTypes = {
  event: PropTypes.shape({
    __typename: PropTypes.string,
    id: PropTypes.string,
    name: PropTypes.string,
    description: PropTypes.string,
    start: PropTypes.string,
    end: PropTypes.string,
    location: PropTypes.string,
  }),
  loading: PropTypes.bool,
};

export default Event;
