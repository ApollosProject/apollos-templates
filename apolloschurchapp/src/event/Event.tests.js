import React from 'react';
import moment from 'moment';

import { Providers, renderWithApolloData } from '@apollosproject/ui-test-utils';

import Event from './Event';

describe('The Event component', () => {
  it('should render', async () => {
    const tree = await renderWithApolloData(
      <Providers>
        <Event
          event={{
            __typename: 'Event',
            id: 'Event:123',
            start: moment('2019-09-26T15:10:51.200Z')
              .utc()
              .toJSON(),
            end: moment('2019-09-26T17:10:51.200Z')
              .utc()
              .toJSON(),
            location: 'Willow Creek, Chicago',
            image: { sources: [{ url: 'https://url.com/image.jpg' }] },
          }}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
