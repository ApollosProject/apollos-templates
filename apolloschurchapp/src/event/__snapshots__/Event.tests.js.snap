// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Event component should render 1`] = `
Array [
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {},
      ]
    }
  />,
  <BVLinearGradient
    colors={
      Array [
        4294967295,
        4294506484,
      ]
    }
    endPoint={
      Object {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      Object {
        "flex": 1,
        "height": "100%",
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RCTScrollView
        onScroll={[Function]}
        scrollEventThrottle={16}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#A5A5A5",
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
            >
              <Image
                fadeDuration={250}
                isLoading={false}
                onLoad={[Function]}
                source={
                  Array [
                    Object {
                      "cache": "force-cache",
                      "uri": "https://url.com/image.jpg",
                      "url": "https://url.com/image.jpg",
                    },
                  ]
                }
                style={
                  Object {
                    "aspectRatio": 1,
                    "backgroundColor": "#A5A5A5",
                    "resizeMode": "cover",
                    "width": "100%",
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 16,
                "paddingVertical": 0,
              }
            }
            vertical={false}
          >
            <Text
              padded={true}
              style={
                Object {
                  "color": "#303030",
                  "fontFamily": "InterUI-Black",
                  "fontSize": 36,
                  "lineHeight": 41.4,
                  "paddingBottom": 20.02,
                  "paddingTop": 24,
                }
              }
            />
            <View
              style={
                Object {
                  "paddingBottom": 18,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "marginBottom": 24,
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={24}
                  bbWidth={24}
                  focusable={false}
                  height={24}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "marginRight": 8,
                      },
                      Object {
                        "opacity": 1,
                      },
                      Object {
                        "flex": 0,
                        "height": 24,
                        "width": 24,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                  width={24}
                >
                  <RNSVGGroup
                    fill={
                      Array [
                        0,
                        4278190080,
                      ]
                    }
                    fillOpacity={1}
                    fillRule={1}
                    font={Object {}}
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    opacity={1}
                    originX={0}
                    originY={0}
                    propList={Array []}
                    rotation={0}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth={1}
                    vectorEffect={0}
                    x={0}
                    y={0}
                  >
                    <RNSVGPath
                      d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm.5-13H11v6l5.25 3.15.75-1.23-4.5-2.67V7z"
                      fill={
                        Array [
                          0,
                          4289045925,
                        ]
                      }
                      fillOpacity={1}
                      fillRule={1}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          0,
                          0,
                        ]
                      }
                      opacity={1}
                      originX={0}
                      originY={0}
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      x={0}
                      y={0}
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
                <View
                  style={
                    Object {
                      "flexDirection": "column",
                    }
                  }
                >
                  <Text
                    bold={true}
                    style={
                      Object {
                        "color": "#303030",
                        "fontFamily": "InterUI-Bold",
                        "fontSize": 16,
                        "lineHeight": 24,
                      }
                    }
                  >
                    Thu, September 26th, 2019
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "#A5A5A5",
                        "fontFamily": "InterUI-Bold",
                        "fontSize": 14,
                        "lineHeight": 18,
                        "marginTop": 4,
                      }
                    }
                  >
                    3:10 PM â€” 5:10 PM
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "marginBottom": 24,
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={24}
                  bbWidth={24}
                  focusable={false}
                  height={24}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "marginRight": 8,
                      },
                      Object {
                        "opacity": 1,
                      },
                      Object {
                        "flex": 0,
                        "height": 24,
                        "width": 24,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                  width={24}
                >
                  <RNSVGGroup
                    fill={
                      Array [
                        0,
                        4278190080,
                      ]
                    }
                    fillOpacity={1}
                    fillRule={1}
                    font={Object {}}
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    opacity={1}
                    originX={0}
                    originY={0}
                    propList={Array []}
                    rotation={0}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth={1}
                    vectorEffect={0}
                    x={0}
                    y={0}
                  >
                    <RNSVGPath
                      d="M11.83 2C8 2 4 4.96 4 9.83c0 4.6 6.96 11.65 7.22 11.9.17.18.35.27.6.27.27 0 .44-.1.6-.26.28-.26 7.23-7.3 7.23-11.9 0-4.88-4-7.84-7.82-7.84zm0 10.43c-1.48 0-2.6-1.13-2.6-2.6 0-1.48 1.12-2.6 2.6-2.6 1.47 0 2.6 1.12 2.6 2.6 0 1.47-1.13 2.6-2.6 2.6z"
                      fill={
                        Array [
                          0,
                          4289045925,
                        ]
                      }
                      fillOpacity={1}
                      fillRule={1}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          0,
                          0,
                        ]
                      }
                      opacity={1}
                      originX={0}
                      originY={0}
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      x={0}
                      y={0}
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
                <View
                  style={
                    Object {
                      "flexDirection": "column",
                    }
                  }
                >
                  <Text
                    bold={true}
                    style={
                      Object {
                        "color": "#303030",
                        "fontFamily": "InterUI-Bold",
                        "fontSize": 16,
                        "lineHeight": 24,
                      }
                    }
                  >
                    Willow Creek, Chicago
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "#A5A5A5",
                        "fontFamily": "InterUI-Bold",
                        "fontSize": 14,
                        "lineHeight": 18,
                        "marginTop": 4,
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "paddingBottom": 18,
                }
              }
            >
              <View />
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </BVLinearGradient>,
]
`;
